spec/models/user_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validations' do
+    context 'handle' do
+      it { is_expected.to validate_length_of(:handle).is_at_least(4).is_at_most(12) }
+      it { is_expected.not_to allow_value('carlo$$$').for(:handle) }
+      it { is_expected.to validate_uniqueness_of(:handle) }
+    end
```

Comment:
You could also check if there's a presence validation and maybe if it allows a valid value <3

```suggestion
    context 'handle' do
      it { is_expected.to validate_presence_of(:handle) }
      it { is_expected.to allow_value('valid_user').for(:handle) }
      it { is_expected.to validate_length_of(:handle).is_at_least(4).is_at_most(12) }
      it { is_expected.not_to allow_value('carlo$$$').for(:handle) }
      it { is_expected.to validate_uniqueness_of(:handle) }
    end
```

Reply:
Yep, I forgot it! Thanks for the reminder!